### DISCUSSION TITLE
scrolling within fixed height container

### DISCUSSION THREAD

**Discussion by shts86** [CarlosNZ/json-edit-react#126] [General]
Hi, I really like your work. 

I tried placing the editor inside a fixed-height container and making it scrollable, but I couldn't find a solution for this in the documentation. I managed to achieve the desired behavior by modifying the CSS externally, adding `height: 100%` and `overflow: auto`, but it feels like a bit of a hack. 

Is there a proper way to accomplish this that I might be missing? Perhaps through the component props?

**Comment by CarlosNZ** [comment]
Hi, yeah, the height is a bit tricky -- the way you've done it is probably the best way I think. I've had to use a somewhat "hacky" solution myself to set the height properties for the collapsing elements, since it's not possible to use a CSS transition when height is set to `auto`. I'm implementing a variation of this solution: https://dev.to/sarah_chima/using-css-transitions-on-the-height-property-al0

Thanks for your comments, sorry for the slow response -- I've been on holiday :) 

BTW, if you end up publishing your implementation online publicly, I'd be keen to check it out -- I've never tried this component as a scrolling element.

**Comment by shts86** [comment]
Hi, I created this [example](https://codesandbox.io/p/sandbox/v7hqsv) for you to check out—it's very simple. I think it would be great to have this as a scrollable prop (or something similar), which would eliminate the need for manually setting the scrolling behavior.
